brickBreaker.js

```diff
@@ -0,0 +1,209 @@
+class BrickBreaker {
+  constructor() {
+    this.canvas = document.getElementById("canvas");
+    this.ctx = this.canvas.getContext("2d");
+    this.initializeEntities();
+    this.initializeEventListeners();
+  }
+
+  initializeEntities() {
+    this.score = 0;
+    this.ball = new Ball(this.ctx, this.canvas.width / 2, 200, 10, "#FFF", 4);
+    this.paddle = new Paddle(
+      this.canvas,
+      this.ctx,
+      100,
+      10,
+      150,
+      525,
+      5,
+      "grey"
+    );
+    this.bricks = [];
+    this.brickRows = 5;
+    this.brickCols = 8;
+    this.brickWidth = 50;
+    this.brickLength = 25;
+
+    const colors = ["red", "blue", "green", "yellow", "purple"];
+
+    for (let col = 0; col < this.brickRows; col++) {
+      for (let row = 0; row < this.brickCols; row++) {
+        if (!this.bricks[row]) {
+          this.bricks[row] = [];
+        }
+        const x = this.brickWidth * row;
+        const y = this.brickLength * col;
+        const brick = new Brick(
+          this.ctx,
+          x,
+          y,
+          this.brickWidth,
+          this.brickLength,
+          colors[col]
+        );
+        this.bricks[row][col] = brick;
+      }
+    }
+  }
+
+  initializeEventListeners() {
+    window.addEventListener("keydown", this.keyDownHandler.bind(this));
+    window.addEventListener("keyup", this.keyUpHandler.bind(this));
+  }
+
+  keyDownHandler(e) {
+    console.log("key", e);
+    switch (e.key) {
+      case "ArrowLeft":
+        this.paddle.setDirection(-1);
+        break;
+      case "ArrowRight":
+        this.paddle.setDirection(1);
+        break;
+      case " ":
+        this.reset();
```

Comment:
you should only reset if the game is at a win/lose state

Reply:
In the other PR, you commented I should have a keypress for reset game. The player should have the option to restart the game whenever they want.

Reply:
most restart games happen when the game is over. if this is what you design it to be though, then so be it